components/renku_data_services/namespace/apispec.py

```diff
@@ -34,7 +34,7 @@ class NamespaceResponse(BaseAPISpec):
         example="my-renku-namespace",
         max_length=99,
         min_length=1,
-        pattern="^[a-z0-9]+[a-z0-9._-]*$",
+        pattern="^(?!.*\\.git$|.*\\.atom$|.*[\\-._][\\-._].*)[a-zA-Z0-9][a-zA-Z0-9\\-_.]*$",
```

Comment:
This same regex appears in many places. Would it be possible to make it a global or at least module-level constant?

Reply:
This is automatically generated code from the apispec.yaml files that we merge from several components. In the apispec.yaml it appears in fewer places.